src/Tribe/Data_API.php

```diff
@@ -0,0 +1,517 @@
+<?php
+
+if ( ! class_exists( 'Tribe__Tickets__Data_API' ) ) {
+
+
+	class Tribe__Tickets__Data_API {
+
+		protected $active_modules;
+		protected $ticket_types = array();
+		protected $ticket_class = array();
+
+		/**
+		 * Class constructor
+		 */
+		public function __construct() {
+			$this->active_modules = Tribe__Tickets__Tickets::modules();
+			$this->setup_data();
+		}
+
+		/**
+		 * Setup activate ticket classes and field for data api
+		 */
+		protected function setup_data() {
+
+			foreach ( Tribe__Tickets__Tickets::modules() as $module_class => $module_instance ) {
+
+				/**
+				 * The usage of plain `$module_class::ATTENDEE_EVENT_KEY` will throw a `T_PAAMAYIM_NEKUDOTAYIM`
+				 * when using PHP 5.2, which is a fatal.
+				 *
+				 * So we have to construct the constant name using a string and use the `constant` function.
+				 */
+				$types['order']   = constant( "$module_class::ORDER_OBJECT" );
+				$types['product'] = $module_class::get_instance()->ticket_object;
+				$types['ticket']  = constant( "$module_class::ATTENDEE_OBJECT" );
+				if ( 'Tribe__Tickets__RSVP' === $module_class ) {
+					$types['ticket'] = $module_class::get_instance()->ticket_object;
+				}
+				$types['attendee'] = constant( "$module_class::ATTENDEE_OBJECT" );
+
+				$this->ticket_class[ $module_class ] = array();
+
+				foreach ( $types as $key => $value ) {
+
+					$this->ticket_types[ $key ][]                = $value;
+					$this->ticket_class[ $module_class ][ $key ] = $value;
+
+				}
+				if ( 'Tribe__Tickets_Plus__Commerce__EDD__Main' === $module_class ) {
+					$this->ticket_class[ $module_class ]['tribe_for_event'] = $module_class::$event_key;
+					//					$this->ticket_class[ $module_class ]['tribe_for_event'] = $module_class::get_instance()->event_key;
```

Comment:
Spacing after `//` is slightly weird - remove or replace with spaces if we need to precisely align with something?